pkg/services/ngalert/api/admin.go

```diff
@@ -0,0 +1,34 @@
+package api
```

Comment:
This file in particular could use simplification. There's no reason for two structs here with the extra layer of indirection.

Reply:
On a second reading, I'd like to do this file naming in a separate PR - there are enough funny things going on with diffs already that it just feels like a bit too much in one change.

I like the idea of using a suffix instead of a prefix (whether that's "facade" or something else) but I also think we could really reduce the number of files here. It makes sense to have an `alertmanager_facade` and an `alertmanager.go` to separate the logic that routes to datasources or to internal code, but for the provisioning code there's no need for two types. So I'd rather reduce the overall file count and just use "provisioning.go", and have special names for alertmanager, ruler and prometheus.